feat: add esx_weather

Kenshiin13 opened the pull request:
### Description

<!-- What does this PR do? Summarize the feature, fix, or improvement. -->

---

### Motivation

<!-- Why are these changes necessary? What problem does it solve? -->

---

### Implementation Details

<!-- How is it implemented? Highlight key technical decisions or logic. -->

---

### Usage Example

```lua
-- Add example usage here
```

---

### PR Checklist

-   [ ] My commit messages and PR title follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard.
-   [ ] My changes have been tested locally and function as expected.
-   [ ] My PR does not introduce any breaking changes.
-   [ ] I have provided a clear explanation of what my PR does, including the reasoning behind the changes and any relevant context.

Kenshiin13 committed:
init

Kenshiin13 committed:
refactor: avoid ipairs

Kenshiin13 committed:
feat: show current zone in nui

Kenshiin13 committed:
...

Kenshiin13 committed:
fix(esx_weather): add weathertypes to nui

Kenshiin13 committed:
refactor(esx_weather): load weathertypes once

Kenshiin13 committed:
feat(esx_weather): add export to toggle sync

Kenshiin13 committed:
refactor(esx_weather): refactor to modular structure

Kenshiin13 committed:
refactor(esx_weather): change event name

Kenshiin13 committed:
feat: add time sync

Kenshiin13 committed:
refactor: use 1 loop

Kenshiin13 committed:
refactor: use guard clauses

Kenshiin13 committed:
refactor: avoid useless obj creation

Kenshiin13 committed:
refactor: add cfg type annotations

Kenshiin13 committed:
refactor: add zone enum

Kenshiin13 committed:
feat: add fxv2 oal

Kenshiin13 committed:
chore: update zone time offsets to be more reasonable

Kenshiin13 committed:
...

Kenshiin13 committed:
chore: disable debug by default

Kenshiin13 committed:
feat: add ToggleTimeSync export

Kenshiin13 committed:
..

simpleC0de committed:
feat: add esx_weather ui

simpleC0de committed:
feat: add modern ui for esx_weather

simpleC0de committed:
fix: remove unnecessary file

Signed-off-by: simpleC0de <hetznecker2@gmail.com>

Kr3mu committed:
Merge pull request #92 from simpleC0de/esx_weather

Esx weather

Zykem committed:
refactor: Use `zoneName` instead of `zone`

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/esx-framework/ESX-Legacy-Addons?pullRequest=88) <br/>All committers have signed the CLA.

Zykem committed:
fix: Apply MS per Game Minute & race condition fix

Used the already existing config value "secondsPerGameMinute".
Fixed a race condition where ByZone would be false.

Zykem committed:
feat: Add time management events & functions

Zykem committed:
fix: return SerializedTime in add function as expected
